ls1intum/Artemis
# [Bugfix] - Refactor Re-evaluate regex usage for text parsing

### Checklist

### Motivation and Context


The prior regex used lookbehinds which work on Chrome etc., but are not supported by Safari.

### Description

We changed the text parsing so it also works on Safari

### Steps for Testing


1. Log in to Artemis
2. Create a short answer quiz (try connecting spots with words before or after, Like `System.[-spot 12].println();`)
3. participate
4. after the quiz endet, click on re-evaluate
5. The text should be correctly divided into boxes (also, every word should be displayed and spots correctly displayed as well)
6. indentations in the original text should also lead to indentations in this particular re-evaluation line (worst word is a bit intended)
6. removing spots and other quiz related actions in this view should work
7. Try with every internet browser you have
